src/api/http.rs

```diff
@@ -102,19 +117,83 @@ fn vdisks(bob: State<BobSrv>) -> Json<Vec<VDisk>> {
 }
 
 #[get("/vdisks/<vdisk_id>")]
-fn vdisk_by_id(bob: State<BobSrv>, vdisk_id: u32) -> Json<Option<VDisk>> {
-    let vdisk = get_vdisk_by_id(&bob, vdisk_id);
-    Json(vdisk)
+fn vdisk_by_id(bob: State<BobSrv>, vdisk_id: u32) -> Option<Json<VDisk>> {
+    get_vdisk_by_id(&bob, vdisk_id).map(Json)
 }
 
 #[get("/vdisks/<vdisk_id>/partitions")]
-fn partitions(_bob: State<BobSrv>, vdisk_id: usize) -> String {
-    format!("partitions of vdisk {}", vdisk_id)
+fn partitions(bob: State<BobSrv>, vdisk_id: u32) -> Result<Json<VDiskPartitions>, Status> {
+    let backend = &bob.grinder.backend.backend();
+    debug!("get backend: OK");
+    let groups = backend.vdisks_groups().ok_or_else(|| {
+        warn!("only pearl backend supports partitions");
+        Status::NotFound
+    })?;
+    debug!("get vdisks groups: OK");
+    let group = groups
+        .iter()
+        .find(|group| group.vdisk_id() == vdisk_id)
+        .ok_or_else(|| {
+            warn!("vdisk with id: {} nont found", vdisk_id);
+            Status::NotFound
+        })?;
+    debug!("group with provided vdisk_id found");
+    let pearls = group.pearls().ok_or_else(|| {
+        error!("writer panics while holding an exclusive lock");
+        Status::InternalServerError
+    })?;
+    debug!("get pearl holders: OK");
+    let pearls: &[_] = pearls.as_ref();
+    let partitions = pearls.iter().map(|pearl| pearl.start_timestamp).collect();
+    let ps = VDiskPartitions {
+        node_name: group.node_name().to_owned(),
+        disk_name: group.disk_name().to_owned(),
+        vdisk_id: group.vdisk_id(),
+        partitions,
+    };
+    trace!("partitions: {:?}", ps);
+    Ok(Json(ps))
 }
 
 #[get("/vdisks/<vdisk_id>/partitions/<partition_id>")]
-fn partition_by_id(_bob: State<BobSrv>, vdisk_id: usize, partition_id: usize) -> String {
-    format!("partition {} of vdisk {}", partition_id, vdisk_id)
+fn partition_by_id(
+    bob: State<BobSrv>,
+    vdisk_id: u32,
+    partition_id: i64,
+) -> Result<Json<Partition>, Status> {
+    let backend = &bob.grinder.backend.backend();
+    debug!("get backend: OK");
+    let groups = backend.vdisks_groups().ok_or_else(|| {
+        warn!("only pearl backend supports partitions");
+        Status::NotFound
```

Comment:
Here should be NotAcceptable (406) status code

Reply:
Also add the same text as you print to log - to statuses: `Status::NotAcceptable("Only pearl backend supports partitions")`